src/include/mutex.i

```diff
@@ -255,3 +259,46 @@ __wt_spin_unlock(WT_SESSION_IMPL *session, WT_SPINLOCK *t)
 #error Unknown spinlock type
 
 #endif
+
+/*
+ * WT_SPIN_INIT_TRACKED --
+ *	Spinlock initialization, with tracking.
+ *
+ * Implemented as a macro so we can pass in a statistics field and convert
+ * it into a slot number.
```

Comment:
I find the use of `slot` confusing here (and in the FIELD_TO_SLOT macro name).  I think of the `slot` as the hashed value you're now storing in `stat_bucket`.  The macro converts field name to stat structure offset.